cparser.nim

```diff
@@ -647,32 +708,37 @@ proc typeAtom(p: var Parser; isTypeDef=false): PNode =
     eat(p, pxParLe, result)
     result.add expression(p)
     eat(p, pxParRi, result)
-  elif isIntType(p.tok.s):
+  elif isBaseIntType(p.tok.s):
     var x = ""
-    #getTok(p, nil)
     var isUnsigned = false
     var isSigned = false
     var isSizeT = false
-    while p.tok.xkind == pxSymbol and (isIntType(p.tok.s) or p.tok.s == "char"):
+    var isDone = false
+    while p.tok.xkind == pxSymbol and (isBaseIntType(p.tok.s) or p.tok.s == "char"):
+      # do a bit more checking to try and handle odd cases like typedef long long someint_t;
       if p.tok.s == "unsigned":
         isUnsigned = true
-      elif p.tok.s == "size_t":
-        isSizeT = true
       elif p.tok.s == "signed":
         isSigned = true
       elif p.tok.s == "int":
-        discard
+        isDone = true
       else:
+        isDone = true
         add(x, p.tok.s)
       getTok(p, nil)
-      if (isSigned or isUnsigned) and p.tok.xkind == pxSymbol and isTypeDef:
-        add(x, p.tok.s)
-        getTok(p, nil)
+      if skipConst(p):
+        isConst = true
+
+    if not isDone and isTypeDef and (isSigned or isUnsigned) and p.tok.xkind == pxSymbol:
+      add(x, p.tok.s)
+      getTok(p, nil)
 
-      if skipConst(p): isConst = true
     if x.len == 0: x = "int"
     let xx = if isSizeT: "csize_t" elif isUnsigned: "cu" & x else: "c" & x
     result = mangledIdent(xx, p, skDontMangle)
+  elif p.tok.s == "size_t":
```

Comment:
this is like a stdint, it requires a special include file - it seems strange to handle this specially, but not the intXX_t types?

Reply:
It’s special cased because of the change in csize vs csize_t I think. 

It actually causes problems here because it’s not an actual “atom type” (e.g. the long long and other types).

I handled the int_t types using a name mangler, which is doesn’t affect parsing type defs.